libbeat/autodiscover/providers/kubernetes/kubernetes_test.go

```diff
@@ -0,0 +1,231 @@
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package kubernetes
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/gofrs/uuid"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/coordination/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	k8sfake "k8s.io/client-go/kubernetes/fake"
+
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+const namespace = "default"
+const leaseName = "metricbeat-lease-test"
+
+// createLease creates a new lease resource
+func createLease() *v1.Lease {
+	lease := &v1.Lease{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      leaseName,
+			Namespace: namespace,
+		},
+	}
+	return lease
+}
+
+// applyLease applies the lease
+func applyLease(client kubernetes.Interface, lease *v1.Lease, firstTime bool) error {
+	var err error
+	if firstTime {
+		_, err = client.CoordinationV1().Leases(namespace).Create(context.Background(), lease, metav1.CreateOptions{})
+		return err
+	}
+	_, err = client.CoordinationV1().Leases(namespace).Update(context.Background(), lease, metav1.UpdateOptions{})
+	return err
+}
+
+// TestLeaseConfigurableFields tests if the leader election is using the fields given in the configuration
+func TestLeaseConfigurableFields(t *testing.T) {
+	client := k8sfake.NewSimpleClientset()
+
+	uuid, err := uuid.NewV4()
+	require.NoError(t, err)
+
+	startLeadingFunc := func(uuid string, eventID string) {}
+	stopLeadingFunc := func(uuid string, eventID string) {}
+	logger := logp.NewLogger("kubernetes-test")
+
+	// the number of leader election managers corresponds to the number of nodes in a cluster with metricbeat
+	var leaseDuration time.Duration
+	var retryPeriod time.Duration
+	var renewDeadline time.Duration
+
+	cfg := Config{
+		Node:          "node-1",
+		LeaderLease:   leaseName,
+		LeaseDuration: 30 * time.Second,
+		RenewDeadline: 25 * time.Second,
+		RetryPeriod:   10 * time.Second,
+	}
+
+	le, err := NewLeaderElectionManager(uuid, &cfg, client, startLeadingFunc, stopLeadingFunc, logger)
+	require.NoError(t, err)
+
+	leaseDuration = le.(*leaderElectionManager).leaderElection.LeaseDuration
+	retryPeriod = le.(*leaderElectionManager).leaderElection.RetryPeriod
+	renewDeadline = le.(*leaderElectionManager).leaderElection.RenewDeadline
+
+	require.Equalf(t, cfg.LeaseDuration, leaseDuration, "lease duration should be the same as the one provided in the configuration.")
+	require.Equalf(t, cfg.RetryPeriod, retryPeriod, "retry period should be the same as the one provided in the configuration.")
+	require.Equalf(t, cfg.RenewDeadline, renewDeadline, "renew deadline should be the same as the one provided in the configuration.")
+}
+
+func TestNewLeaderElectionManager(t *testing.T) {
```

Comment:
could you please add as a comment  regarding the tested scenario for this test?
this comment https://github.com/elastic/beats/pull/38471/files#r1535911348 applies to this test, right? maybe worth mentioning in comment as well

Reply:
Yes, I added a new comment. There are some comments throughout the test that explain what is happening. Is it more understandable now?